packages/tse-backend/src/modules/exchange-data/exchange-data.service.ts

```diff
@@ -0,0 +1,298 @@
+import { ExchangeDataSubscriptionManager } from './subscription-manager.ws.service';
+import { createCompositeKey } from '../../common/utils/subscriptionKey';
+import { MarketDataType } from '../../common/enums/exchange-data.enums';
+import {
+  OHLCVResponse,
+  TickerPriceResponse,
+} from '../../common/interfaces/exchange-data.interfaces';
+
+@Injectable()
+export class ExchangeDataService {
+  private readonly logger = new CustomLogger(ExchangeDataService.name);
+
+  constructor(
+    private readonly exchangeRegistryService: ExchangeRegistryService,
+    @Inject(forwardRef(() => ExchangeDataSubscriptionManager))
+    private readonly subscriptionManager: ExchangeDataSubscriptionManager,
+  ) {}
+
+  async getTickers(command: GetTickersCommand) {
+    const { exchange, symbols } = command;
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance || !exchangeInstance.has.fetchTickers) {
+      throw new Error(
+        'Exchange does not support fetchTickers or is not configured.',
+      );
+    }
+
+    const symbolsArray = symbols.split(',');
+
+    this.logger.log(
+      `Fetching tickers from ${exchangeInstance.name} for ${symbols}`,
+    );
+
+    try {
+      return await exchangeInstance.fetchTickers(symbolsArray);
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  async getOHLCVData(command: GetOHLCVCommand) {
+    const { exchange, symbol, timeframe = '1m', since, limit = 30 } = command;
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance || !exchangeInstance.has.fetchOHLCV) {
+      throw new Error(
+        'Exchange does not support fetchOHLCV or is not configured.',
+      );
+    }
+    this.logger.log(
+      `Fetching OHLCV data from ${exchangeInstance.name} for ${symbol} at ${timeframe} timeframe`,
+    );
+
+    try {
+      const result = await exchangeInstance.fetchOHLCV(
+        symbol,
+        timeframe,
+        since,
+        limit,
+      );
+      return result.map((data) => {
+        return {
+          timestamp: data[0],
+          open: data[1],
+          close: data[2],
+          high: data[3],
+          low: data[4],
+          volume: data[5],
+        } as OHLCVResponse;
+      });
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  async getSupportedPairs() {
+    const exchangeInstances =
+      this.exchangeRegistryService.getSupportedExchanges();
+    const pairs: string[] = [];
+    this.logger.log(
+      `Fetching supported pairs from ${exchangeInstances.length} exchanges`,
+    );
+
+    for (const exchange of exchangeInstances) {
+      const exchangeInstance =
+        this.exchangeRegistryService.getExchange(exchange);
+      if (exchangeInstance && exchangeInstance.has.fetchTickers) {
+        try {
+          const tickers = await exchangeInstance.fetchTickers();
```

Comment:
This should probably be cached if it's an API call, if so add an //TODO comment

Reply:
I agree it should be cached: https://github.com/Hu-Fi/Mr.Market-V2/commit/73d8b9b44ac019d85a18c37a064e3b5f804ea7b6